daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Visual.hs

```diff
@@ -143,14 +147,52 @@ netlistGraph' attrFn outFn assocs = do
 
 
 
-execVisual :: FilePath -> FilePath -> IO ()
-execVisual darFilePath dalfFile = do
-    bytes <- B.readFile dalfFile
-    (_, lfPkg) <- errorOnLeft "Cannot decode package" $ Archive.decodeArchive bytes 
-    world <- darToWorld darFilePath lfPkg
+data Manifest = Manifest { mainDalf :: FilePath , dalfs :: [FilePath] } deriving (Show)
+data ManifestData = ManifestData { mainDalfContent :: BSL.ByteString , dalfsCotent :: [BSL.ByteString] } deriving (Show)
+
+charToWord8 :: Char -> Word8
+charToWord8 = toEnum . fromEnum
+
+cleanString :: String -> String
+cleanString str = T.unpack (T.strip $ T.pack str)
+
+lineToKeyValue :: String -> (String, String)
+lineToKeyValue line = case DLS.splitOn ":" line of
+    [l, r] -> (cleanString l , cleanString r)
+    _ -> ("malformed", "malformed")            
+
+
+manifestMapToManifest :: DHM.HashMap String String -> Manifest
+manifestMapToManifest hash = Manifest mainDalf dependDalfs
+    where
+        mainDalf = DHM.lookupDefault "unknown" "Main-Dalf" hash
+        dependDalfs = map cleanString $ DL.delete mainDalf (DLS.splitOn "," (DHM.lookupDefault "unknown" "Dalfs" hash))
+
+
+manifestDataFromDar :: Archive -> Manifest -> ManifestData
+manifestDataFromDar archive manifest = ManifestData manifestDalfByte dependencyDalfBytes
+    where
+        manifestDalfByte = head [fromEntry e | e <- zEntries archive, ".dalf" `isExtensionOf` eRelativePath e  && eRelativePath e  == mainDalf manifest]
+        dependencyDalfBytes = [fromEntry e | e <- zEntries archive, ".dalf" `isExtensionOf` eRelativePath e  && DL.elem (cleanString (eRelativePath e))  (dalfs manifest)]
+
+manifestFromDar :: Archive -> ManifestData
+manifestFromDar dar =  manifestDataFromDar dar manifest
+    where 
+        manifestEntry = head [fromEntry e | e <- zEntries dar, ".MF" `isExtensionOf` eRelativePath e]
+        lines = BSL.split (charToWord8 '\n') manifestEntry
+        linesStr = map (CH.unpack . BSL.toStrict) lines
+        manifest = manifestMapToManifest $ DHM.fromList $ map lineToKeyValue (filter (\a -> a /= "" ) linesStr)
```

Comment:
Use Data.ByteString.Char8 to avoid having to convert at all. Also use the ByteString lines function.